tests/application_development/code_relocation/linker_arm_nxp_s32.ld

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/linker/sections.h>
+#include <zephyr/devicetree.h>
+
+#include <zephyr/linker/linker-defs.h>
+#include <zephyr/linker/linker-tool.h>
+
+
+#define _SRAM2_DATA_SECTION_NAME  .sram2_data
+#define _SRAM2_BSS_SECTION_NAME   .sram2_bss
+#define _SRAM2_TEXT_SECTION_NAME  .sram2_text
+#define SRAM2_ADDR                (CONFIG_SRAM_BASE_ADDRESS + RAM_SIZE2)
+
+#define RAM_SIZE2 (CONFIG_SRAM_SIZE * 512)
+
+MEMORY
+    {
+    SRAM2       (wx) : ORIGIN = (CONFIG_SRAM_BASE_ADDRESS + RAM_SIZE2), LENGTH = RAM_SIZE2
+
+#ifdef CONFIG_SOC_SERIES_S32K3_M7
+    IVT_HEADER  (r)  : ORIGIN = (CONFIG_FLASH_BASE_ADDRESS + CONFIG_IVT_HEADER_OFFSET),
+                       LENGTH = CONFIG_IVT_HEADER_SIZE
+#endif
+    }
+
+#include <zephyr/arch/arm/aarch32/cortex_m/scripts/linker.ld>
```

Comment:
should be the s32 specific linker file?

Reply:
this is what I mentioned before in https://github.com/zephyrproject-rtos/zephyr/pull/60166#discussion_r1258394185.
The linker file for the Arm test is made in such a way that it doesn't scale if we want to support more SoCs that use SoC-specific linker files (which are most of them). The reason it works for the two boards you ported it, frdm_k64f and mimxrt1060_evk, is because the linker files of those SoCs are just including `#include <zephyr/arch/arm/aarch32/cortex_m/scripts/linker.ld>` (https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nxp_kinetis/k6x/linker.ld, https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nxp_imx/rt/linker.ld). There's nothing special on the NXP S32 platforms with respect of the other Arm SoCs that warrants to use a different linker file as requested.

I think the solution here is to replace `#include <zephyr/arch/arm/aarch32/cortex_m/scripts/linker.ld>` with `#include <linker.ld>` in `linker_arm_sram2.ld`, to extend the SoC-specific linker file and not the arch default linker file.

Reply:
I think this is a good change